lib/binding/kvdb_interface.c

```diff
@@ -453,6 +439,75 @@ hse_kvdb_close(struct hse_kvdb *handle)
     return err;
 }
 
+hse_err_t
+hse_kvdb_attach(
+    const char *kvdb_home_tgt,
+    const char *kvdb_home_src,
+    const char *paths[HSE_MCLASS_COUNT])
+{
+    merr_t         err;
+    char           pidfile_path[PATH_MAX];
+    struct pidfh  *pfh_tgt = NULL, *pfh_src = NULL;
+    size_t         tlen, slen;
+
+    if (HSE_UNLIKELY(!kvdb_home_tgt || !kvdb_home_src)) {
+        log_err("KVDB attach %s home must be provided", kvdb_home_src ? "target" : "source");
+        return merr(EINVAL);
+    }
+
+    tlen = strnlen(kvdb_home_tgt, PATH_MAX);
+    slen = strnlen(kvdb_home_src, PATH_MAX);
+    if (HSE_UNLIKELY(tlen == PATH_MAX || slen == PATH_MAX)) {
+        log_err("KVDB attach %s home path length cannot be longer than PATH_MAX",
+                tlen == PATH_MAX ? "target" : "source");
+        return merr(ENAMETOOLONG);
+    }
```

Comment:
I don't see much utility is using likely/unlikely in code path that aren't hot, just makes the source code noisy..

Reply:
Agreed, will remove.

Reply:
Change this and all the other control plane APIs in this file.